js/VideoPlayer_html5Captionator.js

```diff
@@ -53,25 +53,28 @@ https://source.fluidproject.org/svn/LICENSE.txt
             "text/vtt": "fluid.videoPlayer.html5Captionator.createVttTrack"
         }
     });
-    
-    
+
+
     var bindCaptionatorModel = function (that) {
         var elPaths = that.options.elPaths;
         that.applier.modelChanged.addListener(elPaths.currentCaptions, that.refreshCaptions);
         that.applier.modelChanged.addListener(elPaths.displayCaptions, that.refreshCaptions);
     };
-    
+
     fluid.videoPlayer.html5Captionator.hideAllTracks = function (tracks) {
         fluid.each(tracks, function (trackEl) {
             trackEl.track.mode = "disabled";
         });
     };
-    
+
     fluid.videoPlayer.html5Captionator.showCurrentTrack = function (currentCaptions, tracks, captionSources) {
-        fluid.each(captionSources, function (element, key) {
-            var currentState = $.inArray(key, currentCaptions) === -1 ? "disabled" : "showing";
-            var track = tracks[key].track;
-            track.mode =  track[currentState.toUpperCase()] || currentState;
+        fluid.videoPlayer.html5Captionator.hideAllTracks(tracks);
+        fluid.each(currentCaptions, function (capIndex, index) {
+            var id = captionSources[capIndex].id;
+            var track = fluid.find(tracks, function (track, index) {
+                return (track.id == id) ? track : undefined;
+            });
+            track.track.mode =  "showing";
```

Comment:
I think this could be simplified by using `fluid.byId`. You should be able to just pass in the id to this function and have the correct track element returned. That is, `var track = fluid.byId(id);`. I am also wondering why currentCaptions is an array though. Is it actually possible to have more than one caption showing at the same time?